sql/expression/function/spatial/wkb.go

```diff
@@ -489,6 +526,11 @@ func (l *MLineFromWKB) Type() sql.Type {
 	return types.PolygonType{}
 }
 
+// CollationCoercibility implements the interface sql.CollationCoercible.
+func (*MLineFromWKB) CollationCoercibility(ctx *sql.Context) (collation sql.CollationID, coercibility byte) {
+	return sql.Collation_binary, 4
```

Comment:
I notice some patterns here that aren't expressed.

Particularly, related functions often seem to share coercibility.

What if you created `struct{}` types that represent these classes of related function, implement it once with that type, then embed that type in all those related functions? Does this make the code more self-documenting, or is it not worth it?

Reply:
So that _"related functions often seem to share coercibility"_ part is where the issue lies. For this `FromWKB` line, I can't remember if it is every single one of them, but for another collection of functions that _seem_ like they all return the same results...well they don't. That's the issue with the MySQL collation model, it seems arbitrary. I'd rather someone not implement the functionality at all (and let `GetCoercibility` do something) than unintentionally return the wrong thing but it logically seems to flow that way.